crates/sui-types/src/messages.rs

```diff
@@ -760,13 +765,13 @@ impl SignedTransaction {
     // forming a CertifiedTransaction, where each transaction's authority signature
     // is taking out to form an aggregated signature.
     pub fn to_transaction(self) -> Transaction {
-        Transaction::new(self.data, self.tx_signature)
+        Transaction::new(self.signed_data.data, self.signed_data.tx_signature)
     }
 }
 
 impl Hash for SignedTransaction {
     fn hash<H: Hasher>(&self, state: &mut H) {
-        self.data.hash(state);
+        self.signed_data.data.hash(state);
```

Comment:
Shouldn't we need the whole `signed_data` again (instead of just `data`)?

Reply:
I've added tests for the relevant traits and made changes!